src/ui/Facet/Facet.ts

```diff
@@ -1285,11 +1286,34 @@ export class Facet extends Component {
   protected facetValueHasChanged() {
     this.updateQueryStateModel();
     this.rebuildValueElements();
+    this.clearDependentFacetsIfNoSelectedValuesRemain();
     Defer.defer(() => {
       this.updateAppearanceDependingOnState();
     });
   }
 
+  private clearDependentFacetsIfNoSelectedValuesRemain() {
+    if (this.getSelectedValues().length) {
+      return;
+    }
+
+    this.getDependentFacets().forEach(facet => facet.reset());
+  }
+
+  private getDependentFacets() {
+    const fieldOption = this.options.field;
+    const facets = this.searchInterface.getComponents<Facet>(Facet.ID) || [];
+    const dependentFacets: Facet[] = [];
+
+    facets.forEach(facet => {
```

Comment:
Hum, it's kind of a lot of "wasted" processing everytime a facet value changes, because dependant facet are not used a lot, but this means that all facets must always scan for all others everytime.

I think I would reverse the logic, and make it so the dependant facet is in charge of unselecting it's value when needed, versus making it the parent's duty (because the parents does not know that it is controlling another facet).

Something like this pseudo in the constructor of each facet
```
if(this.dependsOnAnotherfacet()) {
  this.bind.onQueryState('state:change:f@parentFacet', ()=> checkIfParentHasSelectedOrExcludedValueAndUnselectIfNeeded())
}

```